### ISSUE TITLE
dotnet sln add malfunctions on windows

### ISSUE THREAD

**Issue by AlgorithmsAreCool** [dotnet/cli#7699]
This one is pretty weird

## Steps to reproduce
```cmd

cd R:\Dev\repro
dotnet new library --name Fakeone
dotnet new sln

rem Works  (relative-relative)
dotnet sln .\repro.sln add .\fakeone\fakeone.csproj

rem Broken (relative-absolute)
dotnet sln .\repro.sln add r:\dev\repro\fakeone\fakeone.csproj

rem Broken (absolute-relative)
dotnet sln r:\dev\repro\repro.sln add .\fakeone\fakeone.csproj

rem Works (absolute-absolute)
dotnet sln r:\dev\repro\repro.sln add r:\dev\repro\fakeone\fakeone.csproj

rem Broken (absolute-absolute) (exactcase-lowercase)
dotnet sln R:\Dev\repro\repro.sln add r:\dev\repro\fakeone\fakeone.csproj

rem Works (absolute-absolute) (exactcase-exactcase)
dotnet sln R:\Dev\repro\repro.sln add R:\Dev\repro\fakeone\fakeone.csproj
```

## Expected  behavior
The correct project is added to the sln file.
The console emits 
```
Project `fakeone\fakeone.csproj` added to the solution.`
``` 
(note the relative path)

## Actual behavior

The following is added to the sln
```
Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "R:Dev", "R:Dev", "{DFBCFAF7-6159-47BC-8F0C-F0F9A255D0AB}"
EndProject
Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "repro", "repro", "{0A52075C-A2F9-4A66-9E83-9E6BF7D3BECA}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Fakeone", "R:Dev\repro\Fakeone\Fakeone.csproj", "{95E6B721-F2B2-4B57-AF4B-A59F0F4AB69D}"
EndProject
```

and the following is emitted 
```
Project `R:\Dev\repro\fakeone\fakeone.csproj\`
```

## Environment data
`dotnet --info` output:

.NET Command Line Tools (2.0.0)

Product Information:
 Version:            2.0.0
 Commit SHA-1 hash:  cdcd1928c9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d

**Comment by peterhuene** [comment]
Hi @AlgorithmsAreCool:

Thank you for your detailed report!  I have reproduced this issue in `master` and found a few causes: there is a bug that causes drive references without matching case to not be treated as relative and another bug that causes malformed project paths resulting in erroneous solution folders to be added to the solution.  I should have fixes for these issues up for review soon.

**Comment by peterhuene** [comment]
@livarcocc this should have auto-closed with #7842, but I didn't word the commit message correctly.

**Comment by livarcocc** [comment]
Yeah, I can only get it to work with "Fixes <issue_number>". I will close this one then.
